src/server/game/Groups/Group.cpp

```diff
@@ -1875,6 +1875,10 @@ GroupJoinBattlegroundResult Group::CanJoinBattlegroundQueue(Battleground const*
         // don't let join to bg queue random if someone from the group is already in bg queue
         if (bgTemplate->GetBgTypeID() == BATTLEGROUND_RB && member->InBattlegroundQueue())
             return ERR_IN_NON_RANDOM_BG;
+
+        // don't let Death Knights join BG queues when they are not allowed to be teleported yet
+        if (member->getClass() == CLASS_DEATH_KNIGHT && member->GetMapId() == 609 && !member->IsGameMaster() && !member->HasSpell(50977))
+            return ERR_BATTLEGROUND_JOIN_TIMED_OUT;
```

Comment:
This current error is not correct, `ERR_GROUP_JOIN_BATTLEGROUND_FAIL` should be used if I can recall back on retail; if you wanted a true error output.

Reply:
It writes "Join as groupe failed" idk why. I was expecting "%name can't join..." (something like this, can't remember)

But, what are you suggesting then ?

Reply:
Let me research this a bit better and see if I have any other data on this. That error message you used is misleading if not read correctly.